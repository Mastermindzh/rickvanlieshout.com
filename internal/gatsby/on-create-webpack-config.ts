import path from "path";

import { CreateWebpackConfigArgs } from "gatsby";
import { CompilerOptions } from "typescript";

import { compilerOptions } from "../../tsconfig.json";

const onCreateWebpackConfig = (
  (options: Pick<CompilerOptions, "paths">) =>
  ({ actions, getConfig }: CreateWebpackConfigArgs) => {
    // Keep existing TS path aliases in webpack
    actions.setWebpackConfig({
      resolve: {
        alias: Object.entries(options.paths || []).reduce(
          (aliases, [name, [target]]) => ({
            ...aliases,
            [name]: path.resolve(target),
          }),
          {},
        ),
      },
    });

    // Workaround: Gatsby's webpack ESLint plugin is incompatible with ESLint v9+ options
    // on some environments. Remove the plugin from the webpack pipeline during development
    // to prevent build failures like "Invalid Options: Unknown options: extensions, useEslintrc".
    // We still keep linting via npm scripts.
    try {
      const config = getConfig();
      if (config?.plugins?.length) {
        const beforeCount = config.plugins.length;
        config.plugins = config.plugins.filter((plugin: any) => {
          const name = plugin?.constructor?.name;
          return name !== "ESLintWebpackPlugin" && name !== "ESLintPlugin";
        });
        const afterCount = config.plugins.length;
        // Only replace when we've actually modified the plugins array
        if (afterCount !== beforeCount) {
          actions.replaceWebpackConfig(config);
        }
      }
    } catch {
      // noop â€“ if Gatsby changes internals, don't crash the build
    }
  }
)(compilerOptions);

export { onCreateWebpackConfig };
