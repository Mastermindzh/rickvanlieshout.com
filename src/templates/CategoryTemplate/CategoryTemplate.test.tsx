import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import CategoryTemplate from "./CategoryTemplate";
import * as mocks from "@/mocks";


const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("CategoryTemplate", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {
    const props = {
      data: {
        allMarkdownRemark: mocks.allMarkdownRemark,
      },
      pageContext: mocks.pageContext,
    };

    const tree = renderer.create(<CategoryTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
