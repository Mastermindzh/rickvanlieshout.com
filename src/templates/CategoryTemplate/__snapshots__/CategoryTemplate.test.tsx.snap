// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CategoryTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          href="/"
        />
        <h2>
          <a
            href="/"
          >
            Rick van Lieshout
          </a>
        </h2>
        <p />
      </div>
      <span
        className="hideInPrintView"
      >
        <nav
          className="undefined hideInPrintView"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Articles
              </a>
            </li>
            <li>
              <a
                href="/pages/about"
              >
                About Me
              </a>
            </li>
            <li>
              <a
                href="/pages/contacts"
              >
                Contact Me
              </a>
            </li>
          </ul>
        </nav>
        <div>
          <ul>
            <li>
              <a
                href="#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 22 28"
                >
                  <title>
                    rss
                  </title>
                  <path
                    d="M6 21c0 1.656-1.344 3-3 3s-3-1.344-3-3 1.344-3 3-3 3 1.344 3 3zM14 22.922c0.016 0.281-0.078 0.547-0.266 0.75-0.187 0.219-0.453 0.328-0.734 0.328h-2.109c-0.516 0-0.938-0.391-0.984-0.906-0.453-4.766-4.234-8.547-9-9-0.516-0.047-0.906-0.469-0.906-0.984v-2.109c0-0.281 0.109-0.547 0.328-0.734 0.172-0.172 0.422-0.266 0.672-0.266h0.078c3.328 0.266 6.469 1.719 8.828 4.094 2.375 2.359 3.828 5.5 4.094 8.828zM22 22.953c0.016 0.266-0.078 0.531-0.281 0.734-0.187 0.203-0.438 0.313-0.719 0.313h-2.234c-0.531 0-0.969-0.406-1-0.938-0.516-9.078-7.75-16.312-16.828-16.844-0.531-0.031-0.938-0.469-0.938-0.984v-2.234c0-0.281 0.109-0.531 0.313-0.719 0.187-0.187 0.438-0.281 0.688-0.281h0.047c5.469 0.281 10.609 2.578 14.484 6.469 3.891 3.875 6.188 9.016 6.469 14.484z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="mailto:#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 28 28"
                >
                  <title>
                    email
                  </title>
                  <path
                    d="M26 23.5v-12c-0.328 0.375-0.688 0.719-1.078 1.031-2.234 1.719-4.484 3.469-6.656 5.281-1.172 0.984-2.625 2.188-4.25 2.188h-0.031c-1.625 0-3.078-1.203-4.25-2.188-2.172-1.813-4.422-3.563-6.656-5.281-0.391-0.313-0.75-0.656-1.078-1.031v12c0 0.266 0.234 0.5 0.5 0.5h23c0.266 0 0.5-0.234 0.5-0.5zM26 7.078c0-0.391 0.094-1.078-0.5-1.078h-23c-0.266 0-0.5 0.234-0.5 0.5 0 1.781 0.891 3.328 2.297 4.438 2.094 1.641 4.188 3.297 6.266 4.953 0.828 0.672 2.328 2.109 3.422 2.109h0.031c1.094 0 2.594-1.437 3.422-2.109 2.078-1.656 4.172-3.313 6.266-4.953 1.016-0.797 2.297-2.531 2.297-3.859zM28 6.5v17c0 1.375-1.125 2.5-2.5 2.5h-23c-1.375 0-2.5-1.125-2.5-2.5v-17c0-1.375 1.125-2.5 2.5-2.5h23c1.375 0 2.5 1.125 2.5 2.5z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://github.com/#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 26 28"
                >
                  <title>
                    github
                  </title>
                  <path
                    d="M10 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM20 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM22.5 19c0-2.391-1.453-4.5-4-4.5-1.031 0-2.016 0.187-3.047 0.328-0.812 0.125-1.625 0.172-2.453 0.172s-1.641-0.047-2.453-0.172c-1.016-0.141-2.016-0.328-3.047-0.328-2.547 0-4 2.109-4 4.5 0 4.781 4.375 5.516 8.188 5.516h2.625c3.813 0 8.188-0.734 8.188-5.516zM26 16.25c0 1.734-0.172 3.578-0.953 5.172-2.063 4.172-7.734 4.578-11.797 4.578-4.125 0-10.141-0.359-12.281-4.578-0.797-1.578-0.969-3.437-0.969-5.172 0-2.281 0.625-4.438 2.125-6.188-0.281-0.859-0.422-1.766-0.422-2.656 0-1.172 0.266-2.344 0.797-3.406 2.469 0 4.047 1.078 5.922 2.547 1.578-0.375 3.203-0.547 4.828-0.547 1.469 0 2.953 0.156 4.375 0.5 1.859-1.453 3.437-2.5 5.875-2.5 0.531 1.062 0.797 2.234 0.797 3.406 0 0.891-0.141 1.781-0.422 2.625 1.5 1.766 2.125 3.938 2.125 6.219z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://www.twitter.com/#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 26 28"
                >
                  <title>
                    twitter
                  </title>
                  <path
                    d="M25.312 6.375c-0.688 1-1.547 1.891-2.531 2.609 0.016 0.219 0.016 0.438 0.016 0.656 0 6.672-5.078 14.359-14.359 14.359-2.859 0-5.516-0.828-7.75-2.266 0.406 0.047 0.797 0.063 1.219 0.063 2.359 0 4.531-0.797 6.266-2.156-2.219-0.047-4.078-1.5-4.719-3.5 0.313 0.047 0.625 0.078 0.953 0.078 0.453 0 0.906-0.063 1.328-0.172-2.312-0.469-4.047-2.5-4.047-4.953v-0.063c0.672 0.375 1.453 0.609 2.281 0.641-1.359-0.906-2.25-2.453-2.25-4.203 0-0.938 0.25-1.797 0.688-2.547 2.484 3.062 6.219 5.063 10.406 5.281-0.078-0.375-0.125-0.766-0.125-1.156 0-2.781 2.25-5.047 5.047-5.047 1.453 0 2.766 0.609 3.687 1.594 1.141-0.219 2.234-0.641 3.203-1.219-0.375 1.172-1.172 2.156-2.219 2.781 1.016-0.109 2-0.391 2.906-0.781z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="https://t.me/#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 28 28"
                >
                  <title>
                    telegram
                  </title>
                  <path
                    d="M27.563 0.172c0.328 0.234 0.484 0.609 0.422 1l-4 24c-0.047 0.297-0.234 0.547-0.5 0.703-0.141 0.078-0.313 0.125-0.484 0.125-0.125 0-0.25-0.031-0.375-0.078l-7.078-2.891-3.781 4.609c-0.187 0.234-0.469 0.359-0.766 0.359-0.109 0-0.234-0.016-0.344-0.063-0.391-0.141-0.656-0.516-0.656-0.938v-5.453l13.5-16.547-16.703 14.453-6.172-2.531c-0.359-0.141-0.594-0.469-0.625-0.859-0.016-0.375 0.172-0.734 0.5-0.922l26-15c0.156-0.094 0.328-0.141 0.5-0.141 0.203 0 0.406 0.063 0.562 0.172z"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                href="/keys.json"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  viewBox="0 0 20 20"
                >
                  <title>
                    keys
                  </title>
                  <path
                    d="M15 6a1.54 1.54 0 0 1-1.5-1.5 1.5 1.5 0 0 1 3 0A1.54 1.54 0 0 1 15 6zm-1.5-5A5.55 5.55 0 0 0 8 6.5a6.81 6.81 0 0 0 .7 2.8L1 17v2h4v-2h2v-2h2l3.2-3.2a5.85 5.85 0 0 0 1.3.2A5.55 5.55 0 0 0 19 6.5 5.55 5.55 0 0 0 13.5 1z"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
        <div>
          All rights reserved.
        </div>
        <nav
          className="undefined hideInPrintView"
        >
          <ul />
        </nav>
        <span
          className="hideInPrintView"
        >
          Switch color mode:
          <br />
          <div
            className="react-toggle"
            onClick={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="react-toggle-track"
            >
              <div
                className="react-toggle-track-check"
              >
                <div
                  style={
                    {
                      "color": "#f1c40f",
                      "fontSize": "1.1em",
                      "position": "absolute",
                      "top": "-4px",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-moon"
                    data-icon="moon"
                    data-prefix="fas"
                    role="img"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M256 0C114.6 0 0 114.6 0 256S114.6 512 256 512c68.8 0 131.3-27.2 177.3-71.4 7.3-7 9.4-17.9 5.3-27.1s-13.7-14.9-23.8-14.1c-4.9 .4-9.8 .6-14.8 .6-101.6 0-184-82.4-184-184 0-72.1 41.5-134.6 102.1-164.8 9.1-4.5 14.3-14.3 13.1-24.4S322.6 8.5 312.7 6.3C294.4 2.2 275.4 0 256 0z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="react-toggle-track-x"
              >
                <div
                  style={
                    {
                      "color": "#f39c12",
                      "fontSize": "1em",
                      "left": "-3px",
                      "position": "absolute",
                      "top": "-3px",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-sun"
                    data-icon="sun"
                    data-prefix="fas"
                    role="img"
                    viewBox="0 0 576 512"
                  >
                    <path
                      d="M178.2-10.1c7.4-3.1 15.8-2.2 22.5 2.2l87.8 58.2 87.8-58.2c6.7-4.4 15.1-5.2 22.5-2.2S411.4-.5 413 7.3l20.9 103.2 103.2 20.9c7.8 1.6 14.4 7 17.4 14.3s2.2 15.8-2.2 22.5l-58.2 87.8 58.2 87.8c4.4 6.7 5.2 15.1 2.2 22.5s-9.6 12.8-17.4 14.3L433.8 401.4 413 504.7c-1.6 7.8-7 14.4-14.3 17.4s-15.8 2.2-22.5-2.2l-87.8-58.2-87.8 58.2c-6.7 4.4-15.1 5.2-22.5 2.2s-12.8-9.6-14.3-17.4L143 401.4 39.7 380.5c-7.8-1.6-14.4-7-17.4-14.3s-2.2-15.8 2.2-22.5L82.7 256 24.5 168.2c-4.4-6.7-5.2-15.1-2.2-22.5s9.6-12.8 17.4-14.3L143 110.6 163.9 7.3c1.6-7.8 7-14.4 14.3-17.4zM207.6 256a80.4 80.4 0 1 1 160.8 0 80.4 80.4 0 1 1 -160.8 0zm208.8 0a128.4 128.4 0 1 0 -256.8 0 128.4 128.4 0 1 0 256.8 0z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="react-toggle-thumb"
            />
            <input
              checked={false}
              className="react-toggle-screenreader-only"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
          </div>
        </span>
      </span>
    </div>
  </div>
  <div>
    <div>
      <h1>
        typography
      </h1>
      <div>
        <div>
          <div>
            <div>
              <time
                dateTime="September 1, 2016"
              >
                September 1, 2016
              </time>
              <span />
              <span>
                <a
                  href="/typography"
                >
                  typography
                </a>
              </span>
            </div>
            <h2>
              <a
                href="/posts/perfecting-the-art-of-perfection"
              >
                Perfecting the Art of Perfection
              </a>
            </h2>
            <p>
              An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.
            </p>
            <a
              href="/posts/perfecting-the-art-of-perfection"
            />
          </div>
          <div>
            <div>
              <time
                dateTime="September 1, 2016"
              >
                September 1, 2016
              </time>
              <span />
              <span>
                <a
                  href="/design-inspiration"
                >
                  design inspiration
                </a>
              </span>
            </div>
            <h2>
              <a
                href="/posts/the-birth-of-movable-type"
              >
                Johannes Gutenberg: The Birth of Movable Type
              </a>
            </h2>
            <p>
              German inventor Johannes Gutenberg developed a method of movable type and used it to create one of the western world’s first major printed books, the “Forty–Two–Line” Bible.
            </p>
            <a
              href="/posts/the-birth-of-movable-type"
            />
          </div>
        </div>
        <div
          className="undefined"
        >
          <div>
            <a
              className=" hideInPrintView"
              href="/typography/page/1"
              rel="prev"
            >
              ← PREV
            </a>
          </div>
          <div>
            <a
              className=" hideInPrintView"
              href="/typography/page/3"
              rel="next"
            >
              NEXT →
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="CookieConsent"
    style={
      {
        "bottom": "0",
      }
    }
  >
    <div
      className=""
      style={{}}
    >
      This website uses cookies to enhance the user experience.
       
      <small>
        Click 
        <a
          href="/pages/legal/cookies"
        >
          here
        </a>
         to learn more about cookies.
      </small>
    </div>
    <div
      className=""
    >
      <button
        aria-label="Accept cookies"
        className=""
        id="rcc-confirm-button"
        onClick={[Function]}
        style={{}}
      >
        I understand
      </button>
    </div>
  </div>
</div>
`;
